docs/encryption/knative-encryption.md

```diff
@@ -0,0 +1,53 @@
+# Knative Serving Encryption
+
+⛔️ Warning: not everything is implemented, as Knative Serving encryption is still a work-in-progress. Tracking-issue: https://github.com/knative/serving/issues/11906.
+
+## Overview
+There are three parts to Knative Serving encryption
+* (1) HTTPS on the ingress layer _external_ to the cluster (cluster external domain, like `myapp-<namespace>.example.com`).
+* (2) HTTPS on the ingress layer _internal_ to the cluster (cluster local domains, like `myapp.<namespace>.svc.cluster.local`).
+* (3) HTTPS between Knative internal components (`ingress-controller`, `activator`, `queue-proxy`).
+
+📝 Note: currently all control-plane traffic (including Kubernetes PreStopHooks and metadata like metrics) are not (yet) encrypted.
+
+![Overview of Knative encryption](./encryption-overview.drawio.svg)
+
+## The parts in detail
+The different parts are independent of each other and (can) use different Certificate Authorities to sign the necessary certificates.
+
+### (1) External domain
+
+![External domain](./encryption-external-domain.drawio.svg)
+
+* Certificate CN/SAN contains the external domain of a Knative Service, e.g. `myapp-<namespace>.example.com`.
+* The certificates are hosted using SNI by the external endpoint of the ingress-controller.
+* The caller has to trust the (external) CA that signed the certificates (this is out of the scope of Knative).
+* These certificates are either [provided manually](https://knative.dev/docs/serving/using-a-tls-cert/) or by using an implementation to the `Knative Certificate` abstraction. Currently, we support two implementations:
```

Comment:
Maybe either a link to the "Knative Certificate abstraction" in the code (e.g. CRDs) or a brief explanation what this abstraction actually provides would be nice to understand the wider context.

Reply:
Done.